login/services/query/queryLogin.js

```
@@ -0,0 +1,34 @@
+(function(){
+    'use strict';
+    angular.module('query.services', []).service('queryService', [
+        '$http',
+        '$q',
+        queryService
+    ]);
+
+    function queryService($http, $q){
+        queryService.$http = $http;
+        queryService.$q = $q;
+
+        queryService.sendGetRequest = function(user){
+            var promiseResponse = $q;
+            return queryService.$http({
+                method: 'GET',
+                //url: "http://httpbin.org/status/420",
+                //url: "http://httpbin.org/ip",
+                url: "https://httpbin.org/get",
+                data: user,
+                headers: {'Content-Type': 'application/x-www-form-urlencoded'}
+            }).then(function successCallback(httpResponse){
+                return httpResponse.data;
+            },function errorCallback(httpResponse){
+                return promiseResponse.reject(httpResponse);
```

Very good!! this is the same you have to do in the useCase if you want to propagate the error towards the controller.